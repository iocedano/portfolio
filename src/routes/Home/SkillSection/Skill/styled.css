.skill-item-img {
  width: 50px;
}
