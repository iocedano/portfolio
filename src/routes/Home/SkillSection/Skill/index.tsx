import React, { useEffect, useState, memo } from 'react';
import PropTypes, { InferProps } from 'prop-types';
import validIcons from '../../../../config/validicons.json';
import './styled.css';

const SkillPropsTypes = {
  type: PropTypes.oneOf(validIcons.list).isRequired,
  description: PropTypes.string
};

export type SkillType = InferProps<typeof SkillPropsTypes>;

const Skill = ({ type, description } : SkillType) => {
  const [img, setImg] = useState('');

  useEffect(() => {
    import(`../../../../assets/icons/${type}.svg`).then((image) => {
      setImg(image.default);
    });
  }, [type]);

  return (
    <li className="skill-item">
      <div className="skill-item-img-content">
        <img className="skill-item-img" src={img} alt={`skill know job with require ${type} developer`} />
      </div>
      {description && (
        <p className="skill-item-description">{description}</p>
      )}
    </li>
  );
};

Skill.propTypes = SkillPropsTypes;

Skill.defaultProps = {
  description: undefined
}

export default memo(Skill);
