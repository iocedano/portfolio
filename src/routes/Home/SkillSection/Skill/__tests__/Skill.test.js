import React from 'react';
import { render, act } from '@testing-library/react';
import Skill from '..';

describe('Home/SkillSection/Skill', () => {
  it('should render', async () => {
    const promise = Promise.resolve();
    const { container } = render(<Skill type="yarn" />);

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  })
});
