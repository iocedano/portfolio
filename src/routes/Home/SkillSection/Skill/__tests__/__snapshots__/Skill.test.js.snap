// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/SkillSection/Skill should render 1`] = `
<li
  class="skill-item"
>
  <div
    class="skill-item-img-content"
  >
    <img
      alt="skill know job with require yarn developer"
      class="skill-item-img"
      src=""
    />
  </div>
</li>
`;
